include/dlaf/communication/async.h

```diff
@@ -0,0 +1,108 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+/// @file
+
+#include <utility>
+#include <vector>
```

Comment:
I don't see where you use `std::vector` here...is it needed?

Reply:
Good catch, I was using it earlier when prototyping and forgot to remove it.